File: ea-robot-flutter/lib/domain/repositories/robot_repository.dart
import '../entities/robot.dart';

abstract class RobotRepository {
  Future<Robot> getStatus();
  Future<void> connect();
  Future<void> disconnect();
  Future<void> move();
  Future<void> stop();
  Stream<Robot> getTelemetry();
  Future<ThermalState> getNativeThermalState();
}

File: ea-robot-react-native/src/domain/repositories/RobotRepository.ts
import { Robot } from '../entities/Robot';

export interface RobotRepository {
  getStatus(): Promise<Robot>;
  connect(): Promise<Robot>;
  disconnect(): Promise<Robot>;
  move(): Promise<boolean>;
  stop(): Promise<boolean>;
  subscribeToTelemetry(callback: (data: any) => void): () => void;
}
